### DISCUSSION TITLE
What the decrypt method for AES/CTR/NoPadding ?

### DISCUSSION THREAD

**Discussion by ffcactus** [vectordotdev/vrl#291] [Q&A] [answered]
Java client send the encrypted message by using
```Java
import package javax.crypto.Cipher;

val cipher = Cipher.getInstance("AES/CTR/NoPadding");
```

**Comment by fuchsnj** [comment]
It should be one of the following, I'm not sure which one Java uses by default here. It should be easy enough to test with an encrypted payload at https://playground.vrl.dev/ (VRL runs client side here, but please don't submit secrets just to be safe)

```
AES-256-CTR
AES-192-CTR
AES-128-CTR
```

**Reply by fuchsnj** [reply, marked as answer]
None of these might actually work, it seems like Java uses a different "endianness" than VRL currently supports. More info here: https://github.com/vectordotdev/vrl/discussions/297

**Reply by ffcactus** [reply]
Yes, VRL uses a different endian than most of the libraries. I hope it can be fixed soon. Thank you.
